packages/web/src/common/store/notifications/parseAndProcessNotifications.ts

```diff
@@ -144,6 +144,13 @@ export function* parseAndProcessNotifications(
       userIdsToFetch.push(notification.userId)
       trackIdsToFetch.push(notification.entityId)
     }
+    if (
+      type === NotificationType.USDCPurchaseBuyer ||
+      type === NotificationType.USDCPurchaseSeller
+    ) {
+      userIdsToFetch.push(...[...new Set(notification.userIds)])
```

Comment:
This syntax threw me a bit and I had to look up `Set` to find the example of spreading it into an array. Is the goal to deduplicate the userIds?

Reply:
yeaaa i was following existing code in the same file that was de-duping userIds but agree it's confusing and there should be a better way to do this? maybe `...new Set(notification.userIds)` should work?

Reply:
Tried it out locally and `push(...new Set(notification.userIds))` works just fine.
But was also thinking that we have multiple conditionals in this function that are all checking notifications and pushing ids. So across the various notifications we are looping here, we could end up with duplicates anyway.

If you wanted to take a cleanup opportunity, you could:
* Switch the arrays defined on line 49 to `Set`s
* Switch the `push` statements in all of these conditionals to `add`
* At the end, when we call the various retrieve functions, convert the Sets to arrays (`call(retrieveTracks, { trackIds: Array.from(trackIdsToFetch) }`)